src/protectedApi_v8/user/profile-details.ts

```diff
@@ -376,14 +392,42 @@ profileDeatailsApi.post('/createUser', async (req, res) => {
                 sbProfileUpdateReq.profileDetails.personalDetails = _.omit(sbProfileUpdateReq.profileDetails.personalDetails, 'mobile')
             }
 
-            if (req.body.personalDetails.designation) {
-                const arrDesignation = []
-                const objDesignation = {
-                    designation: (req.body.personalDetails.designation) ? req.body.personalDetails.designation :  '',
-                }
-                arrDesignation.push(objDesignation)
-                const profDetailsPropertyName = 'professionalDetails'
-                sbProfileUpdateReq[profDetailsPropertyName] = arrDesignation
+            if (sbdob_ === undefined || sbdob_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.personalDetails = _.omit(sbProfileUpdateReq.profileDetails.personalDetails, 'dob')
+            }
+            if (sbdomicileMedium_ === undefined || sbdomicileMedium_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.personalDetails = _.omit(sbProfileUpdateReq.profileDetails.personalDetails, 'domicileMedium')
+            }
+            if (sbpincode_ === undefined || sbpincode_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.personalDetails = _.omit(sbProfileUpdateReq.profileDetails.personalDetails, 'pinCode')
+            }
+            if (sbgender_ === undefined || sbgender_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.personalDetails = _.omit(sbProfileUpdateReq.profileDetails.personalDetails, 'gender')
+            }
+            if (sbcategory_ === undefined || sbcategory_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.personalDetails = _.omit(sbProfileUpdateReq.profileDetails.personalDetails, 'category')
+            }
+            if (sbdesignation_ === undefined || sbdesignation_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.professionalDetails = _.omit(sbProfileUpdateReq.profileDetails.professionalDetails, 'designation')
+            }
+            if (sbgroup_ === undefined || sbgroup_ === '') {
+                // tslint:disable-next-line: all
+                sbProfileUpdateReq.profileDetails.professionalDetails = _.omit(sbProfileUpdateReq.profileDetails.professionalDetails, 'group')
+            }
```

Comment:
check these are mandatory -- if so, return 400 bad request when the value is not specified

Reply:
Name, Email, Mobile number is mandatory get confirmation form Vinod